main.py

```diff
@@ -134,11 +140,29 @@ def __init__(self):
         self.isBallSelected = False             # Boolean value if a ball is selected 
         self.selectedBall = Ball(0)             # Variable to hold the selected ball object
         self.balls = []                         # List that holds all ball objects in COORDS
+        self.buttons = []                       # List that holds all button objects
+        self.moves = []                         # List that holds the last states to undo
 
         # Append balls[]-list with a new ball for each ball in COORDS
         for coord in COORDS:
             self.balls.append(Ball(len(self.balls)))
 
+        # Add Restart and Undo Button
+        self.buttons.append(Button((120,10,120,40), "Restart", self.__init__))
+        self.buttons.append(Button((260,10,120,40), " Undo", self.undo))
```

Comment:
Nice To Have: "Magic Numbers" könnte man durch Macros ersetzen. Die wirken an dem Punkt vllt im ersten Moment etwas aus der Luft gegriffen, bzw. die vier Zahlen kann man weiter oben direkt initialisieren. Würde ein eventuelles Verschieben der Buttons in möglichen späteren Design-Entscheidungen etwas erleichtern.

Reply:
Oder wir lagern jegliche Macros/Defines in z.B. eine macros.py/defines.py aus und sammeln da alle ohne direkt die main voll zu ballern.

Reply:
Defines für die "Magic Numbers" erstellt. Könnten noch ausgelagert werden, finde das so aber gerade noch übersichtlich genug.